contracts/strategies/fluid/FluidStrategy.sol

```diff
@@ -0,0 +1,209 @@
+        }
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function supportsCollateral(address _asset) external view override returns (bool) {
+        return assetToPToken[_asset] != address(0);
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkBalance(address _asset) external view override returns (uint256) {
+        return allocatedAmount[_asset];
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkAvailableBalance(address _asset) external view override returns (uint256) {
+        uint256 availableLiquidity = _getAvailableLiquidity(_asset);
+        uint256 allocatedValue = allocatedAmount[_asset];
+        if (availableLiquidity <= allocatedValue) {
+            return availableLiquidity;
+        }
+        return allocatedValue;
+    }
+
+    /// @notice Collect accumulated reward token and send to Vault.
+    /// @dev There are no separate rewards by Fluid.
+    function collectReward() external pure override {
+        revert NoRewardIncentive();
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkRewardEarned() external pure override returns (RewardData[] memory) {
+        return (new RewardData[](0));
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkInterestEarned(address _asset) public view override returns (uint256) {
+        uint256 availableLiquidity = _getAvailableLiquidity(_asset);
+        uint256 allocatedValue = allocatedAmount[_asset];
+        if (availableLiquidity > allocatedValue) {
+            return (availableLiquidity - allocatedValue);
+        }
+        return 0;
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkLPTokenBalance(address _asset) public view override returns (uint256) {
+        return IERC20(_getPTokenFor(_asset)).balanceOf(address(this));
+    }
+
+    /// @notice Internal withdraw function used for withdrawing from the strategy.
+    /// @param _recipient Receiver of the funds.
+    /// @param _asset Asset to be withdrawn.
+    /// @param _amount Amount to be withdrawn.
+    /// @return _amount Amount withdrawn.
+    function _withdraw(address _recipient, address _asset, uint256 _amount) internal returns (uint256) {
+        Helpers._isNonZeroAddr(_recipient);
+        Helpers._isNonZeroAmt(_amount, "Must withdraw something");
+
+        // Checking for shares required to be burned to get the desired _amount and checking maximum redeemable shares.
+        address lpToken = _getPTokenFor(_asset);
+        uint256 shares = IFluidToken(lpToken).previewWithdraw(_amount);
+        if (shares > IFluidToken(lpToken).maxRedeem(address(this))) {
+            revert LimitReached();
+        }
+
+        allocatedAmount[_asset] -= _amount;
+
+        // Redeeming the shares.
+        uint256 received = IFluidToken(lpToken).redeem(shares, _recipient, address(this));
+        if (received < _amount) {
```

Comment:
We should use `withdrawSlippage` variable for slippage check @arcantheon.

Reply:
Added deposit and withdrawal slippage check here: 8172a1222815cc9e5829196012183747a56a7b69